src/main.py

```diff
@@ -76,28 +77,34 @@ def run(self) -> None:
             gha_utils.append_job_summary(pull_request_body)
 
             if not self.user_config.skip_pull_request:
-                new_branch_name = f"gh-actions-update-{int(time.time())}"
+                new_branch_name = self.user_config.pull_request_branch
+                if not self.user_config.pull_request_unique:
+                    new_branch_name += f"-{int(time.time())}"
+
                 create_new_git_branch(self.env.base_branch, new_branch_name)
```

Comment:
This function needs to be updated, because it will raise an error if the branch already exists.

Reply:
I was skeptic about this. I made some tests and it seems to work as is.

The git repo used during the action is freshly cloned and the PR branch, if it exists, is only a remote branch.

We should have some thing like that:
```
* main
  remotes/origin/HEAD -> origin/main
  remotes/origin/main
  remotes/origin/gh-actions-update
```

Then, in the function `create_new_git_branch()` the only thing done is:
```
git checkout main
git checkout -b gh-actions-update
```

Which will lead to this (with no error):
```
  main
* gh-actions-update
  remotes/origin/HEAD -> origin/main
  remotes/origin/main
  remotes/origin/gh-actions-update
```

Finally the `git_commit_changes()` function will force push the branch, so the remote branch will be overwritten (with no error).